File: 0374-guess-number-higher-or-lower/0374-guess-number-higher-or-lower.cpp
/** 
 * Forward declaration of guess API.
 * @param  num   your guess
 * @return 	     -1 if num is lower than the guess number
 *			      1 if num is higher than the guess number
 *               otherwise return 0
 * int guess(int num);
 */

class Solution {
public:
    int guessNumber(int n) {
        long long low = 1, high = n, mid, ans = -1;
        while(low <= high) {
            mid = (low + high + 1) >> 1;
            if(guess(mid) == 0) {
                ans = mid;
                break;
            } else if(guess(mid) == -1) {
                high = mid - 1;
            } else {
                low = mid + 1;
            }
        }
        
        return ans;
    }
};

File: 0374-guess-number-higher-or-lower/0374-guess-number-higher-or-lower.java
/** 
 * Forward declaration of guess API.
 * @param  num   your guess
 * @return 	     -1 if num is higher than the picked number
 *			      1 if num is lower than the picked number
 *               otherwise return 0
 * int guess(int num);
 */

public class Solution extends GuessGame {
    public int guessNumber(int n) {
        long low = 1, high = n;
        long mid = -1;
        while(low <= high) {
            mid = (low + high + 1) / 2;
            if(guess((int) mid) == 0)
                return (int) mid;
            else if(guess((int) mid) == -1) {
                high = mid - 1;
            } else {
                low = mid + 1;
            }
        }
        
        return (int) mid;
    }
}